src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -200,9 +200,18 @@ void PollingResolver::GetResultStatus(absl::Status status) {
         gpr_log(GPR_INFO, "[polling resolver %p] retrying immediately", this);
       }
     }
-    Ref(DEBUG_LOCATION, "next_resolution_timer").release();
-    GRPC_CLOSURE_INIT(&on_next_resolution_, OnNextResolution, this, nullptr);
-    grpc_timer_init(&next_resolution_timer_, next_try, &on_next_resolution_);
+    next_resolution_timer_handle_ =
+        event_engine_->RunAfter(timeout, [self = Ref()]() mutable {
+          ApplicationCallbackExecCtx callback_exec_ctx;
+          ExecCtx exec_ctx;
+          auto* self_ptr = static_cast<PollingResolver*>(self.get());
+          self_ptr->work_serializer_->Run(
+              [self = std::move(self)]() {
+                auto* self_ptr = static_cast<PollingResolver*>(self.get());
```

Comment:
This cast in this innermost lambda is unnecessary, RefCountedPtr overrides `operator->`.